model/src/main/kotlin/utils/SimplePackageConfigurationProvider.kt

```diff
@@ -88,6 +60,31 @@ class SimplePackageConfigurationProvider(
         }
 }
 
+/**
+ * A [PackageConfigurationProvider] that provides all [PackageConfiguration]s recursively found in the given directory.
+ * All files in the directory with a [known extension][FileFormat.findFilesWithKnownExtensions] must be package
+ * configuration files, each containing only a single package configuration. Throws an exception if there is more than
+ * one configuration per [Identifier] and [Provenance].
+ */
+class DirectoryPackageConfigurationProvider(directory: File) :
+    SimplePackageConfigurationProvider(readDirectory(directory))
+
+/**
+ * A [PackageConfigurationProvider] that provides all [PackageConfiguration]s found in the given file. Throws an
+ * exception if there is more than one configuration per [Identifier] and [Provenance].
+ */
+class FilePackageConfigurationProvider(file: File) :
+    SimplePackageConfigurationProvider(file.readValue<List<PackageConfiguration>>())
+
+private fun readDirectory(directory: File) =
+    FileFormat.findFilesWithKnownExtensions(directory).map { file ->
+        runCatching {
+            file.readValue<PackageConfiguration>()
+        }.getOrElse { e ->
+            throw IOException("Error reading package configuration from '${file.absolutePath}'.", e)
+        }
+    }
```

Comment:
Nit: For grouping to its (currently) only caller, maybe move this to a companion object inside `DirectoryPackageConfigurationProvider`?

Reply:
I thought we agreed a while ago in a dev meeting that we only move functions into the companion object if there is a technical or design reason to do so, to reduce overall indentation, but to put small helper functions like this on the top level.

Reply:
You're right. It's just situations like this (multiple class declarations in a single file, resulting in unclear "ownership" of the private function) where I personally find our rule to not be very convenient to follow. But let's stick to what we agreed. However, maybe more the `readDirectory` function directly below `DirectoryPackageConfigurationProvider`?

Reply:
Done.